.github/workflows/_buildpacks-prepare-release.yml

```diff
@@ -23,30 +33,41 @@ on:
 jobs:
   prepare-release:
     name: Prepare Release
-    runs-on: pub-hk-ubuntu-22.04-small
+    runs-on: ${{ inputs.ip_allowlisted_runner }}
     steps:
+      - name: Get token for GH application (Linguist)
+        uses: heroku/use-app-token-action@main
+        id: generate-token
+        with:
+          app_id: ${{ inputs.app_id }}
+          private_key: ${{ secrets.app_private_key }}
+
       - name: Checkout
         uses: actions/checkout@v3
+        with:
+          # The token is used to set the local git config which is used to run authenticated git commands
```

Comment:
I think this new comment is missing the key part - which is answering "why do I need a token for a public repo?". Instead the comment states something that IMO can already be seen/figured out by reading the code (that a token is being set for git auth).

In https://github.com/heroku/languages-github-actions/pull/95#discussion_r1283270681, you mention using the token ensures that git signing is set up as expected - so I think the comment needs to say something roughly like:

> "We set the token here even though it's checking out a public repo, because it ensures XYZ is set up as expected for when we commit changes later." 

...where XYZ is something about git signing.